packages/bpk-component-modal/src/BpkModalV2.js

```diff
@@ -0,0 +1,179 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* @flow strict */
+
+import { type Node, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+
+import BpkCloseButton from '../../bpk-component-close-button';
+import { cssModules } from '../../bpk-react-utils';
+
+import STYLES from './BpKModalV2.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+export type Props = {
+  id: string | undefined,
+  ariaLabelledby: string,
+  children: Node,
+  closeLabel: string,
+  fullScreenOnDesktop: boolean,
+  isOpen: boolean,
+  noFullScreenOnMobile: boolean,
+  onClose: (event: SyntheticEvent<>) => void | null,
+  padded: boolean,
+  title: ?string | null,
+  wide: boolean,
+};
+
+type DialogProps = {
+  isDialogOpen: boolean,
+};
+
+const dialogSupported = typeof HTMLDialogElement === 'function';
+
+const setPageProperties = ({ isDialogOpen }: DialogProps) => {
+  document.body.style.overflowY = isDialogOpen ? 'hidden' : 'visible';
+
+  if (!dialogSupported) {
+    document.body.style.position = isDialogOpen ? 'fixed' : 'relative';
+    document.body.style.width = isDialogOpen ? '100%' : 'auto';
+  }
+};
+
+export const BpkModalV2 = (props: Props) => {
+  const {
+    ariaLabelledby,
+    children,
+    closeLabel,
+    fullScreenOnDesktop,
+    id,
+    isOpen,
+    noFullScreenOnMobile,
+    onClose,
+    padded,
+    title,
+    wide,
+  } = props;
+
+  const ref = useRef<HTMLDialogElement>(null);
+
+  useEffect(() => {
+    if (isOpen) {
+      ref.current?.showModal?.();
+
+      const dialog = document.getElementById(id);
+      if (dialog) {
+        dialog.addEventListener('click', (event) =>
+          event.target?.id === id ? ref.current?.close?.() : null,
+        );
+      }
+    } else {
+      ref.current?.close?.();
+    }
+
+    setPageProperties({ isDialogOpen: isOpen });
+    return () => setPageProperties({ isDialogOpen: false });
+  }, [id, isOpen]);
+
+  const classNames = [getClassName('bpk-modal')];
+  const contentClassNames = [getClassName('bpk-modal_container')];
+
+  if (fullScreenOnDesktop) {
+    classNames.push(getClassName('bpk-modal--full-screen-desktop'));
+    contentClassNames.push(
+      getClassName('bpk-modal__container--full-screen-desktop'),
+    );
+  }
+  if (noFullScreenOnMobile) {
+    classNames.push(getClassName('bpk-modal--no-full-screen-mobile'));
+  }
+  if (padded) {
+    contentClassNames.push(getClassName('bpk-modal__container--padded'));
+  }
+  if (wide) {
+    classNames.push(getClassName('bpk-modal--wide'));
+  }
```

Comment:
We can simplify this and make it match our other components via the following:

```suggestion
  const classNames = getClassName(
     'bpk-modal',
     fullScreenOnDesktop && 'bpk-modal--full-screen-desktop',
     noFullScreenOnMobile && 'bpk-modal--no-full-screen-mobile',
     wide && 'bpk-modal--wide'
  );

  const contentClassNames = getClassName(
     'bpk-modal_container',
     fullScreenOnDesktop && 'bpk-modal__container--full-screen-desktop',
     padded && 'bpk-modal__container--padded'
  );
```